[wasm] Fix trimming errors for library tests, to help building with AOT

radical opened the pull request.

radical committed:
[wasm] Reduce verbosity level for debug message

radical committed:
[wasm] Fix trimming errors for System.Runtime.Serialization.Json.Tests

.. and System.Runtime.Serialization.Json.ReflectionOnly.Tests

Partially fixes https://github.com/dotnet/runtime/issues/50730 .

radical committed:
[wasm] Fix trimming errors for System.Runtime.Serialization.Xml.ReflectionOnly.Tests

Partially fixes: https://github.com/dotnet/runtime/issues/50730

radical committed:
Re-enable Serialization tests, from https://github.com/dotnet/runtime/issues/50730

radical committed:
[wasm] Fix trimming errors for System.Runtime.Caching.Tests

Fixes https://github.com/dotnet/runtime/issues/50726

```
fail: [FAIL] System.Runtime.Caching.Tests.AdditionalCacheTests.DisposedCacheTest (TaskId:496)
info: System.Configuration.ConfigurationErrorsException : Configuration system failed to initialize (TaskId:496)
info: ---- System.MissingMethodException : No parameterless constructor defined for type 'System.Configuration.ClientConfigurationHost'. (TaskId:496)
info:    at System.Configuration.ConfigurationManager.EnsureConfigurationSystem() in /Users/radical/dev/r3/src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/ConfigurationManager
info:    at System.Configuration.ConfigurationManager.PrepareConfigSystem() in /Users/radical/dev/r3/src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/ConfigurationManager.cs:li
info:    at System.Configuration.ConfigurationManager.GetSection(String sectionName) in /Users/radical/dev/r3/src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/ConfigurationMana
info:    at System.Runtime.Caching.MemoryCacheStatistics.InitializeConfiguration(NameValueCollection config) in /Users/radical/dev/r3/src/libraries/System.Runtime.Caching/src/System/Runtime/Caching/MemoryCach
info:    at System.Runtime.Caching.MemoryCacheStatistics..ctor(MemoryCache memoryCache, NameValueCollection config) in /Users/radical/dev/r3/src/libraries/System.Runtime.Caching/src/System/Runtime/Caching/Mem
info:    at System.Runtime.Caching.MemoryCache.InitDisposableMembers(NameValueCollection config) in /Users/radical/dev/r3/src/libraries/System.Runtime.Caching/src/System/Runtime/Caching/MemoryCache.cs:line 20
info:    at System.Runtime.Caching.MemoryCache.Init(NameValueCollection config) in /Users/radical/dev/r3/src/libraries/System.Runtime.Caching/src/System/Runtime/Caching/MemoryCache.cs:line 387 (TaskId:496)
info:    at System.Runtime.Caching.MemoryCache..ctor(String name, NameValueCollection config) in /Users/radical/dev/r3/src/libraries/System.Runtime.Caching/src/System/Runtime/Caching/MemoryCache.cs:line 355 (
info:    at System.Runtime.Caching.Tests.AdditionalCacheTests.DisposedCacheTest() in /Users/radical/dev/r3/src/libraries/System.Runtime.Caching/tests/AdditionalCacheTests/AdditionalCacheTests.cs:line 16 (Task
info:    at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture) in /Users/radical/dev/r3/src/mono/System.Private.CoreLib/sr
info: ----- Inner Stack Trace ----- (TaskId:496)
info:    at System.RuntimeType.CreateInstanceMono(Boolean nonPublic, Boolean wrapExceptions) in /Users/radical/dev/r3/src/mono/System.Private.CoreLib/src/System/RuntimeType.Mono.cs:line 1525 (TaskId:496)
info:    at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean wrapExceptions) in /Users/radical/dev/r3/src/mono/System.Private.CoreLib/src/System/RuntimeType.Mono.cs:line 1394 (TaskId:4
info:    at System.Activator.CreateInstance(Type type, Boolean nonPublic, Boolean wrapExceptions) in /Users/radical/dev/r3/src/libraries/System.Private.CoreLib/src/System/Activator.RuntimeType.cs:line 97 (Tas
info:    at System.Activator.CreateInstance(Type type, Boolean nonPublic) in /Users/radical/dev/r3/src/libraries/System.Private.CoreLib/src/System/Activator.RuntimeType.cs:line 87 (TaskId:496)
info:    at System.Configuration.TypeUtil.CreateInstance(Type type) in /Users/radical/dev/r3/src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/TypeUtil.cs:line 147 (TaskId:496)
info:    at System.Configuration.Internal.ConfigSystem.System.Configuration.Internal.IConfigSystem.Init(Type typeConfigHost, Object[] hostInitParams) in /Users/radical/dev/r3/src/libraries/System.Configuratio
info:    at System.Configuration.ClientConfigurationSystem..ctor() in /Users/radical/dev/r3/src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/ClientConfigurationSystem.cs:line 2
info:    at System.Configuration.ConfigurationManager.EnsureConfigurationSystem() in /Users/radical/dev/r3/src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/ConfigurationManager
```

radical committed:
[wasm] Reenable tests that are working with EAT now

radical committed:
[wasm] Fix trimming errors for System.Runtime tests

Specifically `System.Text.Tests.EncodingTests`.

```
fail: [FAIL] System.Text.Tests.EncodingTests.GetEncoding_FromProvider_ByCodePage_WithDisallowedEncoding_Throws(encodingName: "utf-7", codePage: 65000) (TaskId:531)
info: System.ArgumentNullException : Value cannot be null. (Parameter 'method') (TaskId:531)
info:    at Castle.DynamicProxy.Generators.Emitters.SimpleAST.ConstructorInvocationStatement..ctor(ConstructorInfo method, Expression[] args) (TaskId:531)
info:    at Castle.DynamicProxy.Generators.Emitters.CodeBuilders.ConstructorCodeBuilder.InvokeBaseConstructor(ConstructorInfo constructor, ArgumentReference[] arguments) (TaskId:531)
info:    at Castle.DynamicProxy.Generators.InvocationTypeGenerator.CreateConstructor(AbstractTypeEmitter invocation, ProxyGenerationOptions options) (TaskId:531)
info:    at Castle.DynamicProxy.Generators.InvocationTypeGenerator.Generate(ClassEmitter class, ProxyGenerationOptions options, INamingScope namingScope) (TaskId:531)
info:    at Castle.DynamicProxy.Contributors.ClassProxyTargetContributor.BuildInvocationType(MetaMethod method, ClassEmitter class, ProxyGenerationOptions options) (TaskId:531)
info:    at Castle.DynamicProxy.Contributors.ClassProxyTargetContributor.GetInvocationType(MetaMethod method, ClassEmitter class, ProxyGenerationOptions options) (TaskId:531)
info:    at Castle.DynamicProxy.Contributors.ClassProxyTargetContributor.GetMethodGenerator(MetaMethod method, ClassEmitter class, ProxyGenerationOptions options, OverrideMethodDelegate overrideMethod) (TaskId:531)
info:    at Castle.DynamicProxy.Contributors.CompositeTypeContributor.ImplementMethod(MetaMethod method, ClassEmitter class, ProxyGenerationOptions options, OverrideMethodDelegate overrideMethod) (TaskId:531)
info:    at Castle.DynamicProxy.Contributors.CompositeTypeContributor.Generate(ClassEmitter class, ProxyGenerationOptions options) (TaskId:531)
info:    at Castle.DynamicProxy.Generators.ClassProxyGenerator.GenerateType(String name, Type[] interfaces, INamingScope namingScope) (TaskId:531)
info:    at Castle.DynamicProxy.Generators.ClassProxyGenerator.<>c__DisplayClass1_0.<GenerateCode>b__0(String n, INamingScope s) (TaskId:531)
info:    at Castle.DynamicProxy.Generators.BaseProxyGenerator.<>c__DisplayClass33_0.<ObtainProxyType>b__0(CacheKey _) (TaskId:531)
info:    at Castle.Core.Internal.SynchronizedDictionary`2[[Castle.DynamicProxy.Generators.CacheKey, Castle.Core, Version=4.0.0.0, Culture=neutral, PublicKeyToken=407dd0808d44fbdc],[System.Type, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].GetOrAdd(CacheKey key, Func`2 valueFactory) (TaskId:531)
info:    at Castle.DynamicProxy.Generators.BaseProxyGenerator.ObtainProxyType(CacheKey cacheKey, Func`3 factory) (TaskId:531)
info:    at Castle.DynamicProxy.Generators.ClassProxyGenerator.GenerateCode(Type[] interfaces, ProxyGenerationOptions options) (TaskId:531)
info:    at Castle.DynamicProxy.DefaultProxyBuilder.CreateClassProxyType(Type classToProxy, Type[] additionalInterfacesToProxy, ProxyGenerationOptions options) (TaskId:531)
info:    at Castle.DynamicProxy.ProxyGenerator.CreateClassProxyType(Type classToProxy, Type[] additionalInterfacesToProxy, ProxyGenerationOptions options) (TaskId:531)
info:    at Castle.DynamicProxy.ProxyGenerator.CreateClassProxy(Type classToProxy, Type[] additionalInterfacesToProxy, ProxyGenerationOptions options, Object[] constructorArguments, IInterceptor[] interceptors) (TaskId:531)
info:    at Moq.CastleProxyFactory.CreateProxy(Type mockType, IInterceptor interceptor, Type[] interfaces, Object[] arguments) (TaskId:531)
info:    at Moq.Mock`1[[System.Text.Encoding, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].InitializeInstance() (TaskId:531)
info:    at Moq.Mock`1[[System.Text.Encoding, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].OnGetObject() (TaskId:531)
info:    at Moq.Mock.get_Object() (TaskId:531)
info:    at Moq.Mock`1[[System.Text.Encoding, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].get_Object() (TaskId:531)
info:    at System.Text.Tests.EncodingTests.GetEncoding_FromProvider_ByCodePage_WithDisallowedEncoding_Throws(String encodingName, Int32 codePage) in /Users/radical/dev/r3/src/libraries/System.Runtime/tests/System/Text/EncodingTests.cs:line 42 (TaskId:531)
info:    at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture) in /Users/radical/dev/r3/src/mono/System.Private.CoreLib/src/System/Reflection/RuntimeMethodInfo.cs:line 378 (TaskId:531)
```

radical committed:
[wasm] Fix trimming errors for System.Tests.ActivatorTests

```
fail: [FAIL] System.Tests.ActivatorTests.TestingCreateInstanceObjectHandle(assemblyName: "TestLoadAssembly", type: "PublicClassSample", returnedFullNameType: "PublicClassSample", exceptionType: null, returnNul
info: System.TypeLoadException : Could not load type 'PublicClassSample' from assembly 'TestLoadAssembly, Version=1.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. (TaskId:531)
info:    at System.Reflection.RuntimeAssembly.GetType(String name, Boolean throwOnError, Boolean ignoreCase) in /Users/radical/dev/r3/src/mono/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs:li
info:    at System.Activator.CreateInstanceInternal(String assemblyString, String typeName, Boolean ignoreCase, BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationA
info:    at System.Activator.CreateInstance(String assemblyName, String typeName) in /Users/radical/dev/r3/src/libraries/System.Private.CoreLib/src/System/Activator.RuntimeType.cs:line 45 (TaskId:531)
info:    at System.Tests.ActivatorTests.TestingCreateInstanceObjectHandle(String assemblyName, String type, String returnedFullNameType, Type exceptionType, Boolean returnNull) in /Users/radical/dev/r3/src/lib
info:    at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture) in /Users/radical/dev/r3/src/mono/System.Private.CoreLib/src
```

Fixes https://github.com/dotnet/runtime/issues/50981

radical committed:
[wasm] Fix trimming errors for Microsoft.Extensions.Logging.Tests

.. and:

- Microsoft.Extensions.Configuration.FileExtensions
- Microsoft.Extensions.Configuration
- Microsoft.Extensions.FileProviders.Composite
- Microsoft.Extensions.Logging
- Microsoft.Extensions.Primitives

Fixes https://github.com/dotnet/runtime/issues/50709

Example failures:

```
[02:19:11] fail: [FAIL] Microsoft.Extensions.Logging.Test.LoggerFactoryTest.CreateDisposeDisposesInnerServiceProvider
[02:19:11] info: System.ArgumentException : Type Moq.Internals.InterfaceProxy is not valid base type for interface proxy, because it does not have accessible parameterless constructor. Only a non-sealed class with non-private default constructor can be used as base type for interface proxy. Please use some other valid type.
[02:19:11] info:    at Castle.DynamicProxy.Generators.InterfaceProxyWithTargetGenerator.ThrowInvalidBaseType(Type type, String doesNotHaveAccessibleParameterlessConstructor)
[02:19:11] info:    at Castle.DynamicProxy.Generators.InterfaceProxyWithTargetGenerator.EnsureValidBaseType(Type type)
[02:19:11] info:    at Castle.DynamicProxy.Generators.InterfaceProxyWithTargetGenerator.GenerateCode(Type proxyTargetType, Type[] interfaces, ProxyGenerationOptions options)
[02:19:11] info:    at Castle.DynamicProxy.DefaultProxyBuilder.CreateInterfaceProxyTypeWithoutTarget(Type interfaceToProxy, Type[] additionalInterfacesToProxy, ProxyGenerationOptions options)
[02:19:11] info:    at Castle.DynamicProxy.ProxyGenerator.CreateInterfaceProxyTypeWithoutTarget(Type interfaceToProxy, Type[] additionalInterfacesToProxy, ProxyGenerationOptions options)
[02:19:11] info:    at Castle.DynamicProxy.ProxyGenerator.CreateInterfaceProxyWithoutTarget(Type interfaceToProxy, Type[] additionalInterfacesToProxy, ProxyGenerationOptions options, IInterceptor[] interceptors)
[02:19:11] info:    at Moq.CastleProxyFactory.CreateProxy(Type mockType, IInterceptor interceptor, Type[] interfaces, Object[] arguments)
[02:19:11] info:    at Moq.Mock`1[[Microsoft.Extensions.Logging.ILoggerProvider, Microsoft.Extensions.Logging.Abstractions, Version=6.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]].InitializeInstance()
[02:19:11] info:    at Moq.Mock`1[[Microsoft.Extensions.Logging.ILoggerProvider, Microsoft.Extensions.Logging.Abstractions, Version=6.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]].OnGetObject()
[02:19:11] info:    at Moq.Mock.get_Object()
[02:19:11] info:    at Moq.Mock`1[[Microsoft.Extensions.Logging.ILoggerProvider, Microsoft.Extensions.Logging.Abstractions, Version=6.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]].get_Object()
[02:19:11] info:    at Microsoft.Extensions.Logging.Test.LoggerFactoryTest.<>c__DisplayClass20_0.<CreateDisposeDisposesInnerServiceProvider>b__3(IServiceProvider _)
[02:19:11] info:    at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitFactory(FactoryCallSite factoryCallSite, RuntimeResolverContext context)
[02:19:11] info:    at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2[[Microsoft.Extensions.DependencyInjection.ServiceLookup.RuntimeResolverContext, Microsoft.Extensions.DependencyInjection, Version=6.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60],[System.Object, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].VisitCallSiteMain(ServiceCallSite callSite, RuntimeResolverContext argument)
[02:19:11] info:    at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.ResolveService(ServiceCallSite callSite, RuntimeResolverContext context, RuntimeResolverLock lockType, ServiceProviderEngineScope serviceProviderEngine)
[02:19:11] info:    at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite callSite, RuntimeResolverContext context)
[02:19:11] info:    at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2[[Microsoft.Extensions.DependencyInjection.ServiceLookup.RuntimeResolverContext, Microsoft.Extensions.DependencyInjection, Version=6.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60],[System.Object, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].VisitCallSite(ServiceCallSite callSite, RuntimeResolverContext argument)
[02:19:11] info:    at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitIEnumerable(IEnumerableCallSite enumerableCallSite, RuntimeResolverContext context)
[02:19:11] info:    at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2[[Microsoft.Extensions.DependencyInjection.ServiceLookup.RuntimeResolverContext, Microsoft.Extensions.DependencyInjection, Version=6.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60],[System.Object, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].VisitCallSiteMain(ServiceCallSite callSite, RuntimeResolverContext argument)
[02:19:11] info:    at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.ResolveService(ServiceCallSite callSite, RuntimeResolverContext context, RuntimeResolverLock lockType, ServiceProviderEngineScope serviceProviderEngine)
[02:19:11] info:    at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite callSite, RuntimeResolverContext context)
[02:19:12] info:    at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2[[Microsoft.Extensions.DependencyInjection.ServiceLookup.RuntimeResolverContext, Microsoft.Extensions.DependencyInjection, Version=6.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60],[System.Object, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].VisitCallSite(ServiceCallSite callSite, RuntimeResolverContext argument)
[02:19:12] info:    at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, RuntimeResolverContext context)
[02:19:12] info:    at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2[[Microsoft.Extensions.DependencyInjection.ServiceLookup.RuntimeResolverContext, Microsoft.Extensions.DependencyInjection, Version=6.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60],[System.Object, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].VisitCallSiteMain(ServiceCallSite callSite, RuntimeResolverContext argument)
[02:19:12] info:    at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.ResolveService(ServiceCallSite callSite, RuntimeResolverContext context, RuntimeResolverLock lockType, ServiceProviderEngineScope serviceProviderEngine)
[02:19:12] info:    at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite callSite, RuntimeResolverContext context)
[02:19:12] info:    at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2[[Microsoft.Extensions.DependencyInjection.ServiceLookup.RuntimeResolverContext, Microsoft.Extensions.DependencyInjection, Version=6.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60],[System.Object, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].VisitCallSite(ServiceCallSite callSite, RuntimeResolverContext argument)
[02:19:12] info:    at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.Resolve(ServiceCallSite callSite, ServiceProviderEngineScope scope)
[02:19:12] info:    at Microsoft.Extensions.DependencyInjection.ServiceLookup.RuntimeServiceProviderEngine.<>c__DisplayClass1_0.<RealizeService>b__1(ServiceProviderEngineScope p)
[02:19:12] info:    at Microsoft.Extensions.DependencyInjection.ServiceLookup.RuntimeServiceProviderEngine.<>c__DisplayClass1_0.<RealizeService>b__0(ServiceProviderEngineScope scope)
[02:19:12] info:    at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngine.GetService(Type serviceType, ServiceProviderEngineScope serviceProviderEngineScope)
[02:19:12] info:    at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngine.GetService(Type serviceType)
[02:19:12] info:    at Microsoft.Extensions.DependencyInjection.ServiceProvider.GetService(Type serviceType)
[02:19:12] info:    at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetService[ILoggerFactory](IServiceProvider provider)
[02:19:12] info:    at Microsoft.Extensions.Logging.LoggerFactory.Create(Action`1 configure)
[02:19:12] info:    at Microsoft.Extensions.Logging.Test.LoggerFactoryTest.CreateDisposeDisposesInnerServiceProvider()
[02:19:12] info:    at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
```

radical committed:
Merge linker descriptors for Castle, and Moq

radical committed:
[wasm] Fix trimming errors for Microsoft.Extensions.Logging.Test.TraceSourceLoggerTest

```
[03:01:05] fail: [FAIL] Microsoft.Extensions.Logging.Test.TraceSourceLoggerTest.Log_Shoud_Add_Exception_To_Message_Whether_Formatter_Is_Null_Or_Not(shouldFormatterBeNull: True)
[03:01:05] info: Castle.DynamicProxy.InvalidProxyConstructorArgumentsException : Can not instantiate proxy of class: System.Diagnostics.TraceListener.
[03:01:05] info: Could not find a parameterless constructor.
[03:01:05] info:    at Castle.DynamicProxy.ProxyGenerator.CreateClassProxyInstance(Type proxyType, List`1 proxyArguments, Type classToProxy, Object[] constructorArguments)
[03:01:05] info:    at Castle.DynamicProxy.ProxyGenerator.CreateClassProxy(Type classToProxy, Type[] additionalInterfacesToProxy, ProxyGenerationOptions options, Object[] constructorArguments, IInterceptor[] interceptors)
[03:01:05] info:    at Moq.CastleProxyFactory.CreateProxy(Type mockType, IInterceptor interceptor, Type[] interfaces, Object[] arguments)
[03:01:05] info:    at Moq.Mock`1[[System.Diagnostics.TraceListener, System.Diagnostics.TraceSource, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a]].InitializeInstance()
[03:01:05] info:    at Moq.Mock`1[[System.Diagnostics.TraceListener, System.Diagnostics.TraceSource, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a]].OnGetObject()
[03:01:05] info:    at Moq.Mock.get_Object()
[03:01:05] info:    at Moq.Mock`1[[System.Diagnostics.TraceListener, System.Diagnostics.TraceSource, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a]].get_Object()
[03:01:05] info:    at Microsoft.Extensions.Logging.Test.TraceSourceLoggerTest.<>c__DisplayClass2_0.<Log_Shoud_Add_Exception_To_Message_Whether_Formatter_Is_Null_Or_Not>b__0(ILoggingBuilder builder)
[03:01:05] info:    at Microsoft.Extensions.DependencyInjection.LoggingServiceCollectionExtensions.AddLogging(IServiceCollection services, Action`1 configure)
[03:01:05] info:    at Microsoft.Extensions.Logging.Test.TestLoggerBuilder.Create(Action`1 configure)
[03:01:05] info:    at Microsoft.Extensions.Logging.Test.TraceSourceLoggerTest.Log_Shoud_Add_Exception_To_Message_Whether_Formatter_Is_Null_Or_Not(Boolean shouldFormatterBeNull)
[03:01:05] info:    at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
```

radical committed:
[wasm] Fix trimming errors for System.ComponentModel.TypeConverter

radical committed:
[wasm] Fix trimming errors in `System.Xml.RW*` tests

```
[02:19:48] info: System.MissingMethodException : Constructor on type 'OLEDB.Test.ModuleCore.XmlInlineDataDiscoverer' not found.
[02:19:48] info:    at System.RuntimeType.CreateInstanceImpl(BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture)
[02:19:48] info:    at System.Activator.CreateInstance(Type type, BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationAttributes)
[02:19:48] info:    at System.Activator.CreateInstance(Type type, Object[] args)
[02:19:48] info:    at DictionaryExtensions.GetOrAdd[Tuple`2,Object](IDictionary`2 dictionary, Tuple`2 key, Func`1 newValue)
```

Fixes https://github.com/dotnet/runtime/issues/50723 .

radical committed:
[wasm] Fix trimming errors for System.Runtime.Serialization.Xml tests

https://github.com/dotnet/runtime/issues/50720

radical committed:
[wasm] Fix trimming errors for System.Xml.XmlSerializer*

Partially fixes https://github.com/dotnet/runtime/issues/50724

radical committed:
[wasm] Fix trimming errors for System.Reflection.Tests

Don't trim `System.Reflection.TestExe`

Fixes https://github.com/dotnet/runtime/issues/50713

radical committed:
[wasm] Fix trimming errors for System.Reflection.Tests

`IList` implementation on `ArrayList` was getting trimmed

```
[02:19:20] fail: [FAIL] System.Reflection.Tests.TypeInfoTests.IsAssignable(type: typeof(System.Collections.IList), c: typeof(System.Collections.ArrayList), expected: True)
[02:19:20] info: Assert.Equal() Failure
[02:19:20] info: Expected: True
[02:19:20] info: Actual:   False
[02:19:20] info:    at System.Reflection.Tests.TypeInfoTests.IsAssignable(Type type, Type c, Boolean expected)
[02:19:20] info:    at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
```

Partially fixes https://github.com/dotnet/runtime/issues/50715

radical committed:
[wasm] Fix trimming errors in System.Reflection.Tests

```
02:19:20] fail: [FAIL] System.Reflection.Tests.AssemblyTests.AssemblyGetForwardedTypesLoadFailure
[02:19:20] info: Assert.Throws() Failure
[02:19:20] info: Expected: typeof(System.Reflection.ReflectionTypeLoadException)
[02:19:20] info: Actual:   (No exception was thrown)
[02:19:20] info:    at System.Reflection.Tests.AssemblyTests.AssemblyGetForwardedTypesLoadFailure()
[02:19:20] info:    at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
```

The relevant code:

```csharp
Assembly a = typeof(TypeInForwardedAssembly).Assembly;
ReflectionTypeLoadException rle = Assert.Throws<ReflectionTypeLoadException>(() => a.GetForwardedTypes());`
```

This is relying on types forwarded to `UnloadableAssembly`, and
`ForwardedTypesAssembly`, so don't trim those.

Fixes https://github.com/dotnet/runtime/issues/50715

radical committed:
[wasm] Fix build

radical committed:
Fix copying pdbs, and dependent assemblies

radical committed:
[wasm] Fix trimming errors for System.Runtime.Extensions.Tests

Dependent assemblies getting trimmed out.

Partially fixes https://github.com/dotnet/runtime/issues/50724

radical committed:
[wasm] DebuggerSupport=true for tests, as many library tests depend on debugger attributes

eerhardt commented on eng/testing/ildescriptors/ILLink.Descriptor.Castle.xml:
```diff
@@ -0,0 +1,13 @@
+<linker>
```
All these files should have `Descriptors` in the name following the naming pattern set by the linker.

See the docs here:
https://github.com/mono/linker/blob/55bd0ebd00ddbf2cd817314b1cf6824003cfb463/docs/data-formats.md#descriptor-format

eerhardt commented on eng/testing/ildescriptors/ILLink.Descriptor.Serialization.xml:
```diff
@@ -0,0 +1,10 @@
+<linker>
+  <assembly fullname="System.Private.DataContractSerialization">
+    <type fullname="System.Runtime.Serialization.KeyValue`2" />
```
cc @joperezr - is your change going to make it so this is no longer necessary?

eerhardt replied:
In general, I don't think we should be adding `ILLink.Descriptors` files for our own assemblies. Instead we should be fixing our assemblies to be trimmable.

eerhardt commented on eng/testing/ildescriptors/ILLink.Descriptor.Serialization.xml:
```diff
@@ -0,0 +1,10 @@
+    <type fullname="System.Runtime.Serialization.KeyValuePairAdapter`2" />
+  </assembly>
+  <assembly fullname="System.Private.Xml.Linq">
+    <type fullname="System.Xml.Linq.XElement" />
```
Why is this necessary? It shouldn't be.

eerhardt commented on src/libraries/System.Runtime.Serialization.Xml/tests/ILLink.Descriptor.xml:
```diff
@@ -0,0 +1,7 @@
+<linker>
+  <assembly fullname="System.Data.Common">
+    <type fullname="System.Data.DataColumn">
+      <method signature="System.Void .ctor()" />
```
cc @krwq - is your change going to fix this?

eerhardt commented on src/libraries/System.Reflection/tests/ILLink.Descriptor.xml:
```diff
@@ -0,0 +1,7 @@
+<linker>
+  <assembly fullname="System.Private.CoreLib">
+      <type fullname="System.Collections.ArrayList" />
```
Why is this necessary? It shouldn't be.

radical replied:
I was seeing this as:
```
02:19:20] fail: [FAIL] System.Reflection.Tests.TypeInfoTests.IsAssignable(type: typeof(System.Collections.IList), c: typeof(System.Collections.ArrayList), expected: True)
[02:19:20] info: Assert.Equal() Failure
[02:19:20] info: Expected: True
[02:19:20] info: Actual:   False
[02:19:20] info:    at System.Reflection.Tests.TypeInfoTests.IsAssignable(Type type, Type c, Boolean expected)
[02:19:20] info:    at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
```
https://github.com/radical/runtime/commit/3cdf08a0ece59578afabc3ffb661de8f7a07d122

This PR is an attempt to remove the trimming errors, to enable AOT tests to be built. I didn't delve into why the linker wasn't being able to pick up the usage. We could look into that, when I make a clean PR based on `main`.

eerhardt commented on src/libraries/Microsoft.Extensions.Logging/tests/Common/ILLink.Descriptor.xml:
```diff
@@ -0,0 +1,7 @@
+<linker>
+  <assembly fullname="System.Diagnostics.TraceSource">
+    <type fullname="System.Diagnostics.TraceListener">
+        <method signature="System.Void .ctor()" />
```
Do you know why this is necessary? It shouldn't be.

radical committed:
[wasm] fix build error for System.Reflection.Tests

This depends on pdb files, which are included only if
`DebuggerSupport==true`, but that is set to false by default for tests.
Override that for this project.

radical committed:
[wasm] Rename ILLink.Descriptor.xml -> *Descriptors.xml

radical committed:
[wasm] Fix build errors due to trimming in System.Reflection.MetadataLoadContext

The test project explicitly copies, and needs `mscorlib.dll`, which gets
trimmed out with `EnableAggressiveTrimming=true`. Preserve that.

radical committed:
Cleanup trimmer descriptors for System.ComponentModel.TypeConverter.Tests

radical committed:
il descriptors cleanup

radical committed:
[wasm] Disable broken test System.Data.Tests.XmlDataReaderTest.XmlLoadTest

Tracked in https://github.com/dotnet/runtime/issues/51369

radical committed:
[wasm][tests] Fix path to System.Text.Json.Tests for excluding from the test run

radical committed:
[wasm] Disable failing test in System.Xml.Xsl.XslCompiledTransformApi.Tests

radical committed:
Re-enable System.Xml.Xsl.XslCompiledTransformApi.Tests

marek-safar commented on src/libraries/Microsoft.Extensions.Logging/tests/Common/ILLink.Descriptors.xml:
```diff
@@ -0,0 +1,7 @@
+<linker>
+  <assembly fullname="System.Diagnostics.TraceSource">
```
This one is a bit unexpected, do you know where is it used?

eerhardt replied:
Agreed - this shouldn't be necessary because all the ILLink warnings are fixed in this assembly. This might be a product bug somewhere.

eerhardt commented on src/libraries/System.Reflection.MetadataLoadContext/tests/ILLink.Descriptors.xml:
```diff
@@ -0,0 +1,3 @@
+<linker>
+  <assembly fullname="mscorlib" />
```
Is this still necessary with the latest linker? We had a few fixes that went in to preserve facade assemblies.

eerhardt commented on src/libraries/System.Reflection/tests/CoreCLR/System.Reflection.CoreCLR.Tests.csproj:
```diff
@@ -14,5 +14,7 @@
   <ItemGroup>
     <ProjectReference Include="..\TestExe\System.Reflection.TestExe.csproj" />
+
+    <TrimmerRootDescriptor Include="$(MSBuildThisFileDirectory)..\ILLink.Descriptors.CoreCLR.xml" />
```
Can this `ILLink.Descriptors.CoreCLR.xml` file move next to the .csproj that needs it?

eerhardt commented on eng/testing/tests.props:
```diff
@@ -6,6 +6,7 @@
     <TestDependsOn Condition="'$(TestNoBuild)' != 'true' and '$(BuildAllProjects)' != 'true'">Build</TestDependsOn>
     <TestDependsOn>$(TestDependsOn);GenerateRunScript;RunTests</TestDependsOn>
     <VSTestNoLogo>true</VSTestNoLogo>
+    <ILDescriptorsPath>$(MSBuildThisFileDirectory)ildescriptors\</ILDescriptorsPath>
```
This needs a better name. How about?

`<ILLinkDescriptorsPath>$(MSBuildThisFileDirectory)ILLinkDescriptors\</ILLinkDescriptorsPath>`

eerhardt commented:
> <linker>

This file should go in the `ILLinkDescriptors` folder.

---
Refers to: eng/testing/ILLink.Descriptors.TestUtilities.xml:1 in e8cb433. [](commit_id = e8cb4334bc2a5faa11496354dac017a5d04e5355, deletion_comment = False)

eerhardt commented on eng/testing/ildescriptors/ILLink.Descriptors.Serialization.xml:
```diff
@@ -0,0 +1,10 @@
+<linker>
+  <assembly fullname="System.Private.DataContractSerialization">
+    <type fullname="System.Runtime.Serialization.KeyValue`2" />
```
Are these still necessary on the latest main? @joperezr addressed the ILLink warnings in this assembly, so these might be fixed.

If they aren't fixed, we should open a bug to make sure we don't have a product bug here.

joperezr replied:
DCS is annotated now but most of it was annotated as unsafe. It would be good to understand who actually needs this so that we can check why is the type not being preserved and to find out if it is expected or not

eerhardt commented on src/libraries/System.Reflection/tests/ILLink.Descriptors.xml:
```diff
@@ -0,0 +1,7 @@
+<linker>
+  <assembly fullname="System.Private.CoreLib">
+      <type fullname="System.Collections.ArrayList" />
```
This is suprising. Why is this necessary?

eerhardt commented on src/libraries/System.Runtime.Serialization.Xml/tests/ILLink.Descriptors.xml:
```diff
@@ -0,0 +1,7 @@
+<linker>
+  <assembly fullname="System.Data.Common">
+    <type fullname="System.Data.DataColumn">
+      <method signature="System.Void .ctor()" />
```
fyi @krwq again here

radical commented:
@eerhardt This PR was based on another big PR which got merged last night. And I had been working on this over the last few weeks. So, it's definitely possible that some of the changes aren't needed anymore, as you suggested.

I'm going to enable all the tests on separate temporary PR, and then apply the fixes that are still useful, and we can have the current error messages etc. Then I can reset the history here.

marek-safar commented:
I think it'd be better to enable trimming analyzer warnings for this than trying to guess where stuff can break. We could exclude the 3rd party libs to make it less noisy.

eerhardt commented:
> be better to enable trimming analyzer warnings

For our test assemblies? I'm unsure if that would be worth the effort.

marek-safar commented:
> I'm unsure if that would be worth the effort.

Why? Isn't that what this and probably more upcoming PRs will do?

eerhardt commented:
1. Addressing ILLink warnings isn't trivial. It takes quite a bit of effort.
2. The maintenance cost of keeping ILLink warnings out of our tests would be an on-going tax for developers.
3. It doesn't completely solve the problem, because of 3rd party libraries, as noted above.

marek-safar commented:
> The maintenance cost of keeping ILLink warnings out of our tests would be an on-going tax for developers.

Once this PR goes it that will be already the case, every PR touching libraries will have to ensure the tests are not failing with the linker running for the browser's test

radical commented:
> Once this PR goes it that will be the case, every PR touching libraries will have to ensure the tests are not failing with the linker running for the browser's test

It's already enabled on `main`, see https://dev.azure.com/dnceng/public/_build/results?buildId=1104194&view=logs&j=fddee5b2-3265-5a91-40c1-750d81e69f3f .

radical committed:
Merge remote-tracking branch 'origin/main' into build-wasm-aot-helix-test

radical commented:
I have re-tested, and applied all the relevant changes with clean commits in https://github.com/dotnet/runtime/pull/51697 . Some changes weren't needed with the latest `main`.

Closing this in favor of that PR.
